app/lib/tariff_synchronizer/taric_file_name_generator_patched.rb

```diff
@@ -0,0 +1,14 @@
+module TariffSynchronizer
+  # TODO: Remove once Taric has fixed their api
+  class TaricFileNameGeneratorPatched
```

Comment:
this class does not do much, I wonder if we can remove this and add a method in `TariffSynchronizer`

Reply:
I'm going to leave it since the TariffSynchronizer is multi-purpose and we'd end up with three public methods and a bunch of private methods that is probably beyond the scope of this task

Reply:
It would be a halfway-house refactor

Reply:
I've opted to move these into the downloader/reduce the number of seemingly arbitary objects we instantiate. Ultimately I totally agree this is very greedy about what the TariffSynchronizer knows and should be there. For the moment I can kill one evil